### ISSUE TITLE
Jumping to definitions doesn't work when defined like $self->{PAYMENT} = PayPal->new();

### ISSUE THREAD

**Issue by Baetek** [Camelcade/Perl5-IDEA#2344]
Hey, this is a truly incredible project, hats off to you. I am trying to convince my company to switch to this Perl IDE to make our dev team life A TONNE easier, and the company would likely support your project then.

I have a problem though, one big selling point is being able to jump to function definitions with CTRL+Click. 
But a lot of our codebase is like this:

test.pl:
`sub new {
$self->{PAYMENT_METHOD} = PayPal->new();
}

sub transaction {
 $self->{PAYMENT_METHOD}->get_transaction();
}
`
then CTR+Click on get_transaction() shows 'can't find declaration to go to'.  

if we just did something like $paypal = $PayPal->new(); 
then CTRL+Click on $paypal->get_transaction(); does take me to the get_transaction definition in PayPal.pl

The problem is the code base is enormous and most of it uses this style of definitions: 
$something->{sampleAccessor} . 

Would you expect this to work? Is this style just not implemented in the plugin or is it my system?
The Perl interpreter version is 5.32.1 on Linux, and the Perl Target version is v5.26, although switching to other versions doesn't appear to fix it.

Thank you very much!

**Comment by hurricup** [comment]
For now it works only with intermediate variable with explicit type like `my PayPal $paypal = $self->{PAYMENT_METHOD}` or getter in class:
```
sub new {
$self->{PAYMENT_METHOD} = PayPal->new();
}

#@returns PayPal
sub payment_method{
    shift->{PAYMENT_METHOD}
}

sub transaction {
 ...
 $self->payment_method->get_transaction();
}

```
Solving this would be nice, but no estimations so far.

**Comment by zurom** [comment]
Why doesn't it work
```perl
# Cannot find declaration to go to
$self->obj->method(); 
```
but this work fine?
```perl
# no annotations
my $foo = $self->obj;

# declaration is found
$foo->method();
```

Is it depend on amount of `refs` before method?
```perl
$self->ref->ref->ref->ref->ref->method();
```

**Comment by hurricup** [comment]
no, this is expected to work. Can you provide a reproducible sample? And plugin/ide versions.
